azure.go

```diff
@@ -62,6 +63,25 @@ type oidcDiscoveryInfo struct {
 	JWKSURL string `json:"jwks_uri"`
 }
 
+type transporter struct {
+	pluginEnv *logical.PluginEnvironment
+	sender    *http.Client
+}
+
+func (tp transporter) Do(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", userAgent(tp.pluginEnv))
+
+	client := tp.sender
+	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
```

Comment:
Maybe we should add a comment here on why we aren't following redirects:

```suggestion
	// don't attempt redirects so we aren't acting as an unintended network proxy
	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
```

Reply:
Added documentation in latest commit